pyteal/ast/if_.py

```diff
@@ -78,6 +81,11 @@ def __str__(self):
     def type_of(self):
         if self.thenBranch is None:
             raise TealCompileError("If expression must have a thenBranch", self)
+
+        if self.elseBranch is None:
+            # if there is only a thenBranch, it must evaluate to TealType.none
+            require_type(self.thenBranch, TealType.none)
+
         return self.thenBranch.type_of()
```

Comment:
~~nit: do we need to worry about `then` and `else` evaluate to `none` and `uint64`?~~

For anyone who has the same question, look at `else` branch: type is enforced to be equal to then branch.

Reply:
I also believe for line 48/49, we should probably call `then` and `else` to enforce the type check, or developers can write in initialization to bypass such test.

Reply:
Lines 38-45 perform the type check for values given to the constructor. Did you see that, or do you have a different concern?

Reply:
ok great! I overlooked the checking, resolving.